vmdb/app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -216,73 +216,40 @@ def save_hosts_inventory(ems, hashes, target = nil)
     return if hashes.nil?
-    target = ems if target.nil?
-
-    ems.ems_folders(true)
-    deletes = if (target == ems)
-      ems.ems_folders.dup
-    else
-      []
-    end
-
-    self.save_inventory_multi(:ems_folders, EmsFolder, ems, hashes, deletes, :uid_ems, nil, :ems_children)
+    save_inventory_multi(:ems_folders, EmsFolder, ems, hashes, (target.nil? || target == ems),
+                         :uid_ems, nil, :ems_children)
     self.store_ids_for_new_records(ems.ems_folders, hashes, :uid_ems)
   end
 
   def save_ems_clusters_inventory(ems, hashes, target = nil)
     return if hashes.nil?
-    target = ems if target.nil?
-
-    ems.ems_clusters(true)
-    deletes = if (target == ems)
-      ems.ems_clusters.dup
-    else
-      []
-    end
-
-    self.save_inventory_multi(:ems_clusters, EmsCluster, ems, hashes, deletes, :uid_ems, nil, :ems_children)
+    save_inventory_multi(:ems_clusters, EmsCluster, ems, hashes, (target.nil? || target == ems),
+                         :uid_ems, nil, :ems_children)
     self.store_ids_for_new_records(ems.ems_clusters, hashes, :uid_ems)
   end
 
   def save_resource_pools_inventory(ems, hashes, target = nil)
     return if hashes.nil?
-    target = ems if target.nil?
-
-    ems.resource_pools(true)
-    deletes = if (target == ems)
-      ems.resource_pools.dup
-    elsif target.kind_of?(Host)
-      target.all_resource_pools_with_default.dup
-    else
-      []
-    end
-
-    self.save_inventory_multi(:resource_pools, ResourcePool, ems, hashes, deletes, :uid_ems, nil, :ems_children)
+    save_inventory_multi(:resource_pools, ResourcePool, ems, hashes, (target.nil? || target == ems),
+                         :uid_ems, nil, :ems_children)
     self.store_ids_for_new_records(ems.resource_pools, hashes, :uid_ems)
   end
 
   def save_customization_specs_inventory(ems, hashes, target = nil)
-    return if hashes.nil?
-    deletes = ems.customization_specs(true).dup
-    self.save_inventory_multi(:customization_specs, CustomizationSpec, ems, hashes, deletes, :name)
+    save_inventory_multi(:customization_specs, CustomizationSpec, ems, hashes, true, :name)
   end
 
   def save_miq_scsi_targets_inventory(guest_device, hashes)
-    return if hashes.nil?
-    deletes = guest_device.miq_scsi_targets(true).dup
-    self.save_inventory_multi(:miq_scsi_targets, MiqScsiTarget, guest_device, hashes, deletes, :uid_ems, :miq_scsi_luns)
+    save_inventory_multi(:miq_scsi_targets, MiqScsiTarget, guest_device, hashes, true, :uid_ems, :miq_scsi_luns)
   end
 
   def save_miq_scsi_luns_inventory(miq_scsi_target, hashes)
-    return if hashes.nil?
-    deletes = miq_scsi_target.miq_scsi_luns(true).dup
-    self.save_inventory_multi(:miq_scsi_luns, MiqScsiLun, miq_scsi_target, hashes, deletes, :uid_ems)
+    save_inventory_multi(:miq_scsi_luns, MiqScsiLun, miq_scsi_target, hashes, true, :uid_ems)
   end
 
   def save_switches_inventory(host, hashes)
     return if hashes.nil?
```

Comment:
Yeah, well I expected it to be there in every save_*_inventory method (it could happen when you want to partially change inventory, so don't pass anything for the key), but if it gets pushed down that's fine I guess.  I thought it read nice to have an escape clause right at the beginning, because it makes it clear that that's one of the possible inputs.